### PR TITLE
feat: APP-189 post secret link

### PR DESCRIPTION
**Pull request by blushi**
## Description

https://regennetwork.atlassian.net/browse/APP-189?atlOrigin=eyJpIjoiYWRjMDY0MWRmNWQwNGUwYmEzZmRkOThjZTVhNzdiZmYiLCJwIjoiaiJ9

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

From https://deploy-preview-2401--regen-marketplace.netlify.app/post/regen:13toVg8bPSA85BdNcn7GibTNUoFD25q9krJRV7sWr2qPhyySgHMDv1A.rdf?token=7bbd0a3d7fde9293d9d1dcdb5b91284c
1. Create a post which private or has private files or locations (or use an existing one for a project you're an admin of) and go the post page while being logged in, you should have an option to share a secret link from the top right button "...".
2. Log out and use the copied secret link to access the post page, you should see all post info
3. Create a public post (or use an existing one for a project you're an admin of) and go to the post page while being logged in, the secret link option shouldn't be available.
4. Share one of your posts with some private data from the project data stream section

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-marketplace/src/pages/Post/Post.Header.tsx [hunk 1]
```diff
@@ -134,8 +143,10 @@ export const PostHeader = ({
                   setShareSuccessBanner(true);
                 }}
               />
-              {/* <SharePrivateMenuItem />
-              <DeleteMenuItem /> */}
+              {!publicPost && (
+                <SharePrivateMenuItem onClick={sharePrivateLink} />
```
Comment (blushi) [inline web-marketplace/src/pages/Post/Post.Header.tsx:147]: TODO add SharePrivateMenuItem to data stream PostCard once https://github.com/regen-network/regen-web/pull/2390 is merged

**Comment by blushi** [comment]
@erikalogie @clevinson see testing instructions 1 to 3.
will ping you once 4 is ready

**Comment by blushi** [comment]
> LGTM! I sent you a few comments on slack about the color of the three dot icon and a new share secret icon that is all one color

@erikalogie updated
You can also test 4. (from your project data stream section)

**Comment by erikalogie** [comment]
This works well! Will your replace the share icon with this one so that the share and secret link icons are the same shade of green?
Share

**Comment by blushi** [comment]
> This works well! Will your replace the share icon with this one so that the share and secret link icons are the same shade of green? Share

updated the existing icon color, no need to replace it entirely

**Review by r41ph — APPROVED** [review]
tACK

**Pull request merged**